marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    mapping.from(Controller.Standard.RTClick).to(function (value) {
+        if (value === 0) {
+            return;
+        }
+        
+        var uuid = getUUIDFromLaser(Controller.Standard.RightHand);
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    // #endregion
+    // *************************************
+    // STOP MAPPING FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START SOLO FUNCTIONS
+    // *************************************
+    // #region Solo
+
+
+    // Adds avatar to the solo list
+    function addSolo(targetUUID) {
+        Audio.addToSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove Avatar from the solo list
+    function removeSolo(targetUUID) {
+        Audio.removeFromSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove all avatars from the solo list
+    function resetSolo() {
+        Audio.resetSoloList();
+        removeAllOverlays();
+        soloAvatars = {};
+        updateUI();
+    }
+
+    
+    // Add an avatar to the list and give them an overlay
+    function addAvatarToList(avatarUUID, displayUsername) {
+        soloAvatars[avatarUUID] = {
+            id: avatarUUID,
+            name: displayUsername
+        };
+
+        addSolo(avatarUUID);
+        addOverlayToUser(avatarUUID);
+    }
+
+
+    // Remove an avatar from the list and remove their overlay
+    function removeAvatarFromList(avatarUUID) {
+        removeOverlay(avatarUUID);
+        delete soloAvatars[avatarUUID];
+        removeSolo(avatarUUID);
+    }
+
+
+    // Handles avatar being solo'd
+    function soloAvatar(avatarUUID) {
+        log("in soloAvatar", avatarUUID);
+
+        log("soloAvatars", soloAvatars);
+
+        if (!isEnabled) {
+            return;
+        }
+
+        var getAvatarClicked = AvatarList.getAvatar(avatarUUID);
+        var displayUsername = getAvatarClicked.sessionDisplayName;
+
+        log("avatar clicked", displayUsername);
+
+        if (soloAvatars[avatarUUID]) {
+            removeAvatarFromList(avatarUUID);
+        } else {
+            addAvatarToList(avatarUUID, displayUsername);
+        }
+    }
+
+
+    // #endregion
+    // *************************************
+    // STOP SOLO FUNCTIONS
+    // *************************************
+
+    // *************************************
+    // START OVERLAY FUNCTIONS
+    // *************************************
+    // #region Overlay
+
+    // Adds a speaker overlay above a user solo'd
+    function addOverlayToUser(uuid) {
+        var user = soloAvatars[uuid];
+        var overlayPosition = AvatarList.getAvatar(uuid).getNeckPosition(); // user.currentPosition
+
+        var WHITE = [255, 255, 255];
```

Comment:
why define this var but all of the others are defined inline? i recommend removing this var and changing line 311